# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: caps/nsScriptSecurityManager.h
# Commit: 955256297d6d
# Full Hash: 955256297d6dc83d52eb77a202c9c320622c0d13
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-11-05 16:16:12
# Regressor Bug: 1579367
# File Overlap Count: 1
# Description:
#   Bug 1579367 - Initialize XPCJSContext explicitly, after loading user prefs. r=kmag
#   
#   This way we get the correct values for start-up prefs in the parent process.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D51061
# ==============================================================================

diff -r 68e7fb2885b0 -r 955256297d6d caps/nsScriptSecurityManager.h
--- a/caps/nsScriptSecurityManager.h	Tue Nov 05 08:19:10 2019 +0200
+++ b/caps/nsScriptSecurityManager.h	Tue Nov 05 06:29:13 2019 +0000
@@ -51,6 +51,9 @@
   // Invoked exactly once, by XPConnect.
   static void InitStatics();
 
+  void InitJSCallbacks(JSContext* aCx);
+  void ClearJSCallbacks(JSContext* aCx);
+
   static already_AddRefed<mozilla::SystemPrincipal>
   SystemPrincipalSingletonConstructor();
 
@@ -121,7 +124,6 @@
 
   static nsIIOService* sIOService;
   static nsIStringBundle* sStrBundle;
-  static JSContext* sContext;
 };
 
 #endif  // nsScriptSecurityManager_h__